src/stream.ts

```diff
@@ -131,6 +102,30 @@ export class WebRTCStream implements Stream {
       let err = (evt as RTCErrorEvent).error;
       this.abort(err);
     };
+
+    const self = this;
+    // reader pipe
+    this.channel.onmessage = async ({data}) => {
+      const res = new Uint8Array(data as ArrayBuffer);
+      if (res.length == 0) {
+        return
+      }
+      this._innersrc.push(res)
+    };
+
+    this._src = pipe(
```

Comment:
this little snippet is cute but it seems needlessly complex. Or am I wrong?
Perhaps I am missing some subtle things not making me appreciate its beauty for what it is.